AppDB/appscale/datastore/taskqueue_client.py

```diff
@@ -0,0 +1,91 @@
+""" Performs TaskQueue service enqueue requests. """
+
+import sys
+
+import requests
+from requests.exceptions import ConnectionError, HTTPError
+
+from appscale.common.constants import TASKQUEUE_SERVICE_PORT
+from appscale.common.unpackaged import APPSCALE_PYTHON_APPSERVER
+
+sys.path.append(APPSCALE_PYTHON_APPSERVER)
+from google.appengine.api.taskqueue import taskqueue_service_pb
+from google.appengine.ext.remote_api import remote_api_pb
+
+
+class EnqueueError(Exception):
+  """ Indicates that the tasks could not be enqueued. """
+  pass
+
+
+class TaskQueueClient(object):
+  """ Performs TaskQueue service enqueue requests. """
+  def __init__(self, locations):
+    """ Creates a new TaskQueueClient.
+
+    Args:
+      locations: A list of IP addresses specifying TaskQueue service locations.
+    """
+    self._locations = [':'.join([ip, str(TASKQUEUE_SERVICE_PORT)])
+                       for ip in locations]
+
+  def add_tasks(self, project_id, service_id, version_id, add_requests):
+    """ Makes a call to the TaskQueue service to enqueue tasks.
+
+    Args:
+      project_id: A string specifying the project ID.
+      service_id: A string specifying the service ID.
+      version_id: A string specifying the version ID.
+      add_requests: A list of TaskQueueAddRequest messages.
+    Raises:
+      EnqueueError if unable to enqueue tasks.
+    """
+    request = taskqueue_service_pb.TaskQueueBulkAddRequest()
+    for add_request in add_requests:
+      request.add_add_request().MergeFrom(add_request)
+
+    api_request = remote_api_pb.Request()
+    api_request.set_method('BulkAdd')
+    api_request.set_service_name('taskqueue')
+    api_request.set_request(request.Encode())
+
+    encoded_api_request = api_request.Encode()
+    headers = {'ProtocolBufferType': 'Request',
+               'AppData': project_id,
+               'Module': service_id,
+               'Version': version_id}
+    api_response = None
+    for location in self._locations:
+      url = 'http://{}'.format(location)
+      try:
+        response = requests.post(url, data=encoded_api_request,
```

Comment:
As it's new code, could it be not blocking from the very beginning?
I see, the only user of this method is run in separate thread, and it's not a big performance improvement. But I would vote for making new function asynchronous where it doesn't require big effort.

Reply:
Yes, good point. I'll replace the requests usage with Tornado's async client.